apps/noise_aware.cpp

```diff
@@ -69,42 +65,89 @@ int main(int argc, char** argv) {
+        ddsim->setNoiseEffects(vm["noise_effects"].as<std::string>());
+        ddsim->setAmplitudeDampingProbability(vm["noise_prob"].as<double>());
+        ddsim->stoch_confidence = vm["confidence"].as<double>();
+        ddsim->setRecordedProperties(vm["properties"].as<std::string>());
+        ddsim->stoch_error_margin = vm["error_bound"].as<double>();
+        ddsim->stochastic_runs    = vm["stoch_runs"].as<long>();
 
-    const std::map<std::string, double> measurement_results = ddsim->StochSimulate();
+        auto t1 = std::chrono::steady_clock::now();
 
-    auto t2 = std::chrono::steady_clock::now();
+        const std::map<std::string, double> measurement_results = ddsim->StochSimulate();
 
-    std::chrono::duration<float> duration_simulation = t2 - t1;
+        auto t2 = std::chrono::steady_clock::now();
 
-    nl::json output_obj;
+        std::chrono::duration<float> duration_simulation = t2 - t1;
 
-    if (vm.count("ps")) {
-        output_obj["statistics"] = {
-                {"simulation_time", duration_simulation.count()},
-                {"benchmark", ddsim->getName()},
-                {"stoch_runs", ddsim->stochastic_runs},
-                {"threads", ddsim->max_instances},
-                {"n_qubits", +ddsim->getNumberOfQubits()},
-                {"applied_gates", ddsim->getNumberOfOps()},
-                {"max_nodes", ddsim->getMaxNodeCount()},
-                {"seed", ddsim->getSeed()},
-        };
+        nl::json output_obj;
 
-        for (const auto& item: ddsim->AdditionalStatistics()) {
-            output_obj["statistics"][item.first] = item.second;
+        if (vm.count("ps")) {
+            output_obj["statistics"] = {
+                    {"simulation_time", duration_simulation.count()},
+                    {"benchmark", ddsim->getName()},
+                    {"stoch_runs", ddsim->stochastic_runs},
+                    {"threads", ddsim->max_instances},
+                    {"n_qubits", +ddsim->getNumberOfQubits()},
+                    {"applied_gates", ddsim->getNumberOfOps()},
+                    {"max_nodes", ddsim->getMaxNodeCount()},
+                    {"seed", ddsim->getSeed()},
+            };
+
+            for (const auto& item: ddsim->AdditionalStatistics()) {
+                output_obj["statistics"][item.first] = item.second;
+            }
         }
-    }
 
-    if (vm.count("pm")) {
-        output_obj["measurement_results"] = measurement_results;
+        if (vm.count("pm")) {
+            output_obj["measurement_results"] = measurement_results;
+        }
+        std::cout << std::setw(2) << output_obj << std::endl;
+    } else if (vm["stoch_runs"].as<long>() < 0) {
+        std::unique_ptr<DeterministicNoiseSimulator> ddsim = std::make_unique<DeterministicNoiseSimulator>(quantumComputation, seed);
+
+        if (vm["stoch_runs"].as<long>() == -2) {
+            ddsim->weird_value_i_dont_understand_but_something_with_sequential = true;
```

Comment:
Maybe change the name of this variable? I didn't understand that it does ^^
Also, `stoch_runs == -2` is not documented in the boost program options.

Reply:
Changed the name to sequentialNoiseApplication and added documentation in the boost program options.